src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java

```diff
@@ -0,0 +1,188 @@
+package edu.cornell.kfs.pdp.batch.service.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.apache.commons.io.FileUtils;
+import org.apache.log4j.Logger;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.core.api.util.type.KualiInteger;
+import org.kuali.kfs.pdp.businessobject.AchAccountNumber;
+import org.kuali.kfs.pdp.businessobject.PaymentDetail;
+import org.kuali.kfs.pdp.businessobject.PaymentGroup;
+import org.kuali.kfs.pdp.businessobject.PaymentStatus;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.Bank;
+
+import com.rsmart.kuali.kfs.pdp.service.AchBundlerHelperService;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Calendar;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+
+public class CuExtractPaymentServiceImplTest {
+    
+    private static final String ACH_BATCH_TEST_FILE_PATH = "src/test/java/edu/cornell/kfs/pdp/batch/service";
+    private static final String ACH_EXTRACT_TEST_FILE_PATH = ACH_BATCH_TEST_FILE_PATH + "/outputFiles";
+    private static final String ACH_EXTRACT_TEST_FILE_NAME = "testAchExtractPaymentsOuput.txt";
+    private static final String ACH_EXTRACT_TEST_PATH_AND_FILE = ACH_EXTRACT_TEST_FILE_PATH + File.separator + ACH_EXTRACT_TEST_FILE_NAME;
+    private static final String ACH_EXTRACT_TEST_PATH_AND_FILE_FOR_VALIDATION = ACH_EXTRACT_TEST_PATH_AND_FILE + ".READY";
+    private static final SimpleDateFormat yyyyMMddHHmmss_DATE_FORMATTER = new SimpleDateFormat("yyyyMMddHHmmss");
+    private static final SimpleDateFormat yyyyMMdd_DATE_FORMATTER = new SimpleDateFormat("yyyyMMdd");
+    private static final String[] NOTIFICATION_EMAIL_ADDRESSES = {"abc@xyz.edu", "def@xyz.edu", "ghi@xyz.edu"};
+    private static final int HEADER_LINE_IN_OUTPUT_FILE = 1;
+    private static final int FIRST_PAYEE_LINE_IN_OUTPUT_FILE = HEADER_LINE_IN_OUTPUT_FILE + NOTIFICATION_EMAIL_ADDRESSES.length + 1;
+    private static final  Logger LOG = org.apache.log4j.Logger.getLogger(CuExtractPaymentServiceImplTest.class);
+    
```

Comment:
Not something I'd require in and of itself, but since there are other requested changes, I'd remove a couple of these blank lines since they seem extraneous.